code/modules/robotics/bot/guardbot.dm

```diff
@@ -406,47 +435,56 @@
 	emag_act(var/mob/user, var/obj/item/card/emag/E)
 		if(!user || !E) return 0
 
+		src.shotcount = 2
 		if (src.idle || !src.on)
-			boutput(user, "You show \the [E] to [src]! There is no response.")
-		else
+			if (!src.emagged)
+				boutput(user, "You wave \the [E] in front of [src]'s blank screen. It responds with a small puff of smoke.")
+			else
+				boutput(user, "You wave \the [E] in front of [src]'s blank screen. It doesn't seem to respond.")
+		else if (!src.emagged)
 			if (E.icon_state == "gold")
 				boutput(user, "You show \the [E] to [src]! They are super impressed!")
 				SPAWN_DBG(1 SECOND)
 					boutput(user, "Like, really REALLY impressed.  They probably think you're some kind of celebrity or something.")
-					sleep(1 SECOND)
-					boutput(user, "Or the president. The president of space.")
+					SPAWN_DBG(1 SECOND)
+						boutput(user, "Or the president. The president of space.")
+						SPAWN_DBG(1 SECOND)
+							boutput(user, "In fact they're so impressed that it shorts out their Spacelaw circuits![pick("", " Whoops.")]")
 			else
-				boutput(user, "You show \the [E] to [src]! They are very impressed.")
-		if (obeygunlaw)
+				boutput(user, "You show \the [E] to [src]! They become so impressed that [pick("they start smelling like burnt circuitry", "you hear a small pop come from inside their casing")].")
+		else
+			boutput(user, "You show \the [E] to [src]! They give you a knowing grin.")
+			set_emotion("smug")
+			update_icon()
+		src.emagged = 1
+		if (src.obeygunlaw)
 			src.obeygunlaw = 0
-			boutput(user, "[src] looks confused for a moment.")
-			src.set_emotion("look")
+			if (src.idle || !src.on)
+				SPAWN_DBG(1 SECOND)
+					boutput(user, "[src] looks confused for a moment.")
+					set_emotion("look")
+		if (src.budgun)
+			if(istype(src.budgun, /obj/item/gun/energy/lawbringer))
+				BeTheLaw(1, 0)
+			if(istype(src.budgun, /obj/item/gun/energy/egun))
+				CheckSafety(src.budgun, 1, user)
 		return 1
 
 	attackby(obj/item/W as obj, mob/user as mob)
 		if (istype(W, /obj/item/device/pda2) && W:ID_card)
 			W = W:ID_card
 		if (istype(W, /obj/item/card/id))
-			if (src.allowed(user))
+			if (src.gunlocklock)
+				speak(pick("Pass.", "No thanks.", "Nah, I'd rather not.", "Hands off the merchandise!",\
+				"Yeah I'm going to need a signed permission slip from your mother first",\
+				"No way, you'll hurt [pick("me", "yourself")]!", "No nerds allowed!",\
+				"You're not the boss of me!", "Couldn't even if I wanted to!"))
+			else if ((src.allowed(user) || user?.mind.assigned_role == "Clown") && !src.gunlocklock)
```

Comment:
really? clown has access to the RD's buddies? Please no.

Reply:
Not anymore they don't!